homeassistant/components/sharkiq/vacuum.py

```diff
@@ -94,14 +95,18 @@ def clean_spot(self, **kwargs: Any) -> None:
         """Clean a spot. Not yet implemented."""
         raise NotImplementedError()
 
-    def send_command(
+    async def async_send_command(
         self,
         command: str,
         params: dict[str, Any] | list[Any] | None = None,
         **kwargs: Any,
     ) -> None:
-        """Send a command to the vacuum. Not yet implemented."""
-        raise NotImplementedError()
+        """Send a command with params to the SharkIq library."""
+        if command == "CLEAN_ROOMS":
+            await self.sharkiq.async_clean_rooms(params)
+            await self.coordinator.async_refresh()
```

Comment:
IMHO, this doesn't make much sense. Why not create a dedicated service for this? Using the commands isn't really end-user-friendly and is actually meant for sending "any" command (no documented/featured by HA).

../Frenck

Reply:
Ah, I guess I just didn't know a) how to add an independent service, and that b) it was ok to go out of the normal "Vacuum" contract.  I had looked at a couple other vacuums and it seemed like the only way to get those to clean specific rooms was to use the sendCommand service.

I'll take a look at making a new service for this.

Reply:
@jrlambs [take a look at this PR](https://github.com/home-assistant/core/pull/89350).  It does the same thing, but uses the method that @frenck suggested.  I just haven't had time to address comments yet.

If yours gets to a point where it's ready to merge first then I'm happy to discard my PR and approve yours.

Feel free to reach out directly if you have questions.  My email is on my profile, and I'm also on the HA discord.

Reply:
@frenck , any idea how I can get the "Rooms" to show up int he UI when setting up the service?
![image](https://github.com/home-assistant/core/assets/13246477/dc5ed422-966d-4661-ad60-9201999db1b8)

I have the service set up in the init to take rooms as a parm:
![image](https://github.com/home-assistant/core/assets/13246477/93086947-19c8-49bc-ac9a-f34eca848759)

I have the param set up in strings and in translations:
![image](https://github.com/home-assistant/core/assets/13246477/d1d92e7e-a694-4b8c-b7cc-3ed03c94cb00)


Somehow the UI doesn't see it though.  I tried using fan_speed as an example but there's enough noise with that search term I'm somehow still missing the key spot to make the UI take that param! Any help is appreciated!

Reply:
Awesome thx for hint, will try this too for the ROMY (waiting currently for review changes approve),
i created 3 extra entity (switch,sensor, buttons) to handle all the stuff like a button for each room (did not create pull request, cause main vacuum branch got not merged yet)

if the vacuum itself could have room support like that would be even better

Reply:
@funkybunch your PR is WAY better.  it actually gets the list of rooms to use.. I was just going to make the service take a list of rooms as a csv string.  I'm going to pull down your PR and see about addressing the comments on it.

Reply:
@frenck @funkybunch, I've basically just copied funkybunch' PR to mine and addressed the comments on the PR while doing so.  

I'm not sure what the correct protocol for helping someone on their PR is.  I'd love to close this one and just push changes directly to the other, but I think I'd need @funkybunch to give me some permissions or something.  

I've updated the documentation on this PR to point to the same documentation PR that was on the original PR

Reply:
@jrlambs sorry for the delay - I added you as a collaborator on my fork.  You should be able to clone and push directly to https://github.com/funkybunch/core/tree/funkybunch/SharkIQ-Room-Support now.  Let me know if you hit any snags.  I'm on the Home Assistant Discord as @funkybunch if you need to ping me directly.

Reply:
Ah, sure I'll take a look at this today and try to move my changes over to your branch and close this PR out.